[CON-168] Update cNodeEndpointToSpIdMap on a cron

theoilie opened the pull request:
### Description
Adds a new queue (`c-node-to-endpoint-sp-id-map-queue`) that reads the chain to update the cNodeEndpoint->spId mapping every 10 seconds locally and every 24 hours on staging/prod. This mapping should almost never change on prod, but locally it needs to be fetched more often because when CN1 is brought up and initially fetches the mapping, it's not aware of CN2/3/4. So we have to re-fetch shortly after CN2/3/4 are brought online.


### Tests
- Updated tests to pass
- Checked queue locally from `/health/bull` endpoint to make sure it was running the jobs at the correct interval and successfully returning the correct mapping


### Monitoring - How will this change be monitored? Are there sufficient logs / alerts?
Check the `c-node-endpoint-to-sp-id-map-queue` queue from the `/health/bull` endpoint to make sure it's running the jobs at the correct interval on staging/prod (24 hours) and successfully returning the correct mapping in the job output.

SidSethi commented on creator-node/src/services/stateMachineManager/index.js:
```diff
@@ -58,15 +48,43 @@ class StateMachineManager {
     }
   }
 
+  async updateMapOnMapFetchJobComplete(jobId, resultString) {
```
nice - can you document

SidSethi reviewed (approved):
looks great! can you add some docs

also just to confirm - did you observe this change to fix the local issues on startup?

any relevant metrics or logs worth adding here? can you also add log strings to Monitoring section

theoilie commented:
> looks great! can you add some docs
> 
> also just to confirm - did you observe this change to fix the local issues on startup?
> 
> any relevant metrics or logs worth adding here? can you also add log strings to Monitoring section

added some docs, updated Monitoring section of PR, and confirmed that I'm not seeing any more incorrect replica set update attempts on local startup 👍 I think we're OK without metrics since this updates so infrequently and can be checked easily on the Bull dashboard if there's a concern about it breaking something upstream (which other metrics would flag) -- wdyt?

SidSethi commented:
> > looks great! can you add some docs
> > also just to confirm - did you observe this change to fix the local issues on startup?
> > any relevant metrics or logs worth adding here? can you also add log strings to Monitoring section
> 
> added some docs, updated Monitoring section of PR, and confirmed that I'm not seeing any more incorrect replica set update attempts on local startup 👍 I think we're OK without metrics since this updates so infrequently and can be checked easily on the Bull dashboard if there's a concern about it breaking something upstream (which other metrics would flag) -- wdyt?

agreed, ship it
